### PR TITLE
Write utility documentation

### PR DESCRIPTION
**Pull request by jlaehne**
### Description of the change
Filled the different proposed sections of `utilities.rst`

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [x] Check whether it is displayed correctly,
- [x] added line to CHANGELOG.md,
- [x] ready for review.

### CODE REVIEW

**Comment by jlaehne** [comment]
@jordiferrero would be nice if you could provide a brief documentation on ``solve_grating_equation`` as a follow up.

**Comment by jlaehne** [comment]
The build documentation can be found here: https://lumispy--128.org.readthedocs.build/en/128/user_guide/utilities.html

No idea why the last run failed on the hyperspy dependencies, but changes were minor and should not affect the way it appears notably.

**Review by jordiferrero — APPROVED** [review]

File: doc/source/user_guide/utilities.rst [hunk 1]
```diff
@@ -5,30 +5,113 @@ Utility functions
...
+
+
+.. _grating_equation-label:
 
+Solving the grating equation
+============================
 
-TODO: Add :py:func:`~.utils.axes.solve_grating_equation`, :py:meth:`~.signals.common_luminescence.CommonLumi.crop_edges`,
-:py:meth:`~.signals.common_luminescence.CommonLumi.remove_negative`?
+TODO: Add a paragraph about the functionality of
```
Comment (jordiferrero) [inline doc/source/user_guide/utilities.rst:116]: For now could you write the following:

The function `solve_grating_equation` follows the conventions described in the tutorial from [Horiba Scientific](https://horiba.com/uk/scientific/products/optics-tutorial/wavelength-pixel-position).

**Comment by jordiferrero** [comment]
> @jordiferrero would be nice if you could provide a brief documentation on `solve_grating_equation` as a follow up.

Yes I should. It is in my todo list. Basically it follows the [Horiba equiations](https://horiba.com/uk/scientific/products/optics-tutorial/wavelength-pixel-position). If you accept this one, then I can open a new PR soon with a more detailed explanation (need to get @Gunnar help). 
I guess for now check my comment.

**Pull request merged**